pkg/filepos/position.go

```diff
@@ -122,3 +123,32 @@ func (p *Position) IsNextTo(otherPostion *Position) bool {
 	}
 	return false
 }
+
+// PopulateAnnotationPositionFromNode takes an annotation's position, the position of the node it annotates,
+// and the comments from the same node, and creates an approximation of the annotation file and line string.
+func PopulateAnnotationPositionFromNode(annPos *Position, nodePosition *Position, nodeComments []Meta) *Position {
+	leftPadding := 0
+	if nodePosition.IsKnown() {
+		nodeLine := nodePosition.GetLine()
+		leftPadding = len(nodeLine) - len(strings.TrimLeft(nodeLine, " "))
+	}
+
+	lineString := ""
+	for _, c := range nodeComments {
+		if c.Position.IsKnown() && c.Position.AsIntString() == fmt.Sprintf("%d", annPos.LineNum()) {
+			lineString = fmt.Sprintf("%v#%s", strings.Repeat(" ", leftPadding), c.Data)
+		}
+	}
+
+	annPos.SetFile(nodePosition.GetFile())
+	annPos.SetLine(lineString)
+
+	return annPos
+}
+
+// Meta is a representation of comment's source, contains comment as a string, and original position.
+// Used to populate
+type Meta struct {
+	Data     string
+	Position *Position
+}
```

Comment:
In my head, the responsibility of the `filepos` package is to provide the domain concept of "Position" in `ytt`.

A "position" in `ytt` uniquely identifies a single line of text. It also provides the means of presenting that "coordinate" to the end-user in various formats.
More recently, it's also provided a means of rendez-vous'ing back with the literal source line (currently, caching it).

Currently, this package is utterly unaware of how this concept is used.

This change is about an algorithm _about_ position, but I'm dubious as to whether it belongs _in_ `filepos`.

With _this_ change, we're expanding the conceptual scope of `filepos` to include ideas like "Node" and "Comment" and "Meta". Those concepts _have_ homes already.

Is there an even better home for this logic?

Reply:
On second thought... this shaped code reads like a kind of "repair" — it's populating a value that "should have" been there from the start...??

... are we discovering that _this_ feature requires more of our YAML parser?

Namely, that _it_ should be populating _even annotations_ with position?

Reply:
Pulling this work into another PR.